tck/api/src/main/java/org/eclipse/microprofile/metrics/tck/TagsTest.java

```diff
@@ -22,53 +22,168 @@
-
-        Assert.assertNotNull(metadata2);
-        Assert.assertTrue(metadata2.getTags().containsKey("size"));
-        Assert.assertTrue(metadata2.getTags().containsValue("green"));
-        Assert.assertFalse(metadata2.getTags().containsValue("blue"));
-        Assert.assertTrue(metadata2.getTags().containsKey("number"));
+    @InSequence(2)
+    public void lastTagValueTest() {
+        
+        Tag tagColour = new Tag("colour","red");
+        Tag tagColourTwo = new Tag("colour","blue");
+        
+        String counterName = "org.eclipse.microprofile.metrics.tck.TagTest.counter";
+        Counter counter = registry.counter(counterName, tagColour, tagColourTwo);
+        
+        //MetricID that only has colour=blue... the last tag value to be passed in
+        MetricID counterMID = new MetricID(counterName, tagColourTwo);
+        
+        //check the metric is registered
+        assertThat("Counter is not registered correctly", registry.getCounters(), hasKey(counterMID));
+    }
+    
+    @Test
+    @InSequence(3)
+    public void counterTagsTest() {
+        
+        Tag tagEarth = new Tag("planet", "earth");
+        Tag tagRed = new Tag("colour", "red");
+        Tag tagBlue = new Tag("colour", "blue");
+        
+        String counterName = "org.eclipse.microprofile.metrics.tck.TagTest.counterColour";
+        
+        Counter counterColour = registry.counter(counterName);
+        Counter counterRed = registry.counter(counterName,tagEarth,tagRed);
+        Counter counterBlue = registry.counter(counterName,tagEarth,tagBlue);
+        
+        MetricID counterColourMID = new MetricID(counterName);
+        MetricID counterRedMID = new MetricID(counterName, tagEarth,tagRed);
+        MetricID counterBlueMID = new MetricID(counterName, tagEarth,tagRed);
+        
+        //check multi-dimensional metrics are registered
+        assertThat("Counter is not registered correctly", registry.getCounters(), hasKey(counterColourMID));
+        assertThat("Counter is not registered correctly", registry.getCounters(), hasKey(counterRedMID));
+        assertThat("Counter is not registered correctly", registry.getCounters(), hasKey(counterBlueMID));
+    }
+    
+    @Test
+    @InSequence(4)
+    public void meterTagsTest() {
+        
+        Tag tagEarth = new Tag("planet", "earth");
+        Tag tagRed = new Tag("colour", "red");
+        Tag tagBlue = new Tag("colour", "blue");
+        
+        String meterName = "org.eclipse.microprofile.metrics.tck.TagTest.meterColour";
+        
+        Meter meterColour = registry.meter(meterName);
+        Meter meterRed = registry.meter(meterName,tagEarth,tagRed);
+        Meter meterBlue = registry.meter(meterName,tagEarth,tagBlue);
+        
+        MetricID meterColourMID = new MetricID(meterName);
+        MetricID meterRedMID = new MetricID(meterName, tagEarth,tagRed);
+        MetricID meterBlueMID = new MetricID(meterName, tagEarth,tagRed);
```

Comment:
think this last one was supposed to say...
MetricID counterBlueMID = new MetricID(counterName, tagEarth,tagBlue);